src/nunit.analyzers/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzer.cs

```diff
@@ -53,47 +83,235 @@ private static void AnalyzeAttribute(SyntaxNodeAnalysisContext context)
-                        var stringConstant = literal.Token.ValueText;
                         context.ReportDiagnostic(Diagnostic.Create(
                             considerNameOfDescriptor,
-                            literal.GetLocation(),
+                            syntaxNode.GetLocation(),
                             stringConstant));
                     }
-                    else
+
+                    if (!symbol.IsStatic)
                     {
-                        var literalText = literal.Token.ValueText;
-                        context.ReportDiagnostic(Diagnostic.Create(missingSourceDescriptor, literal.GetLocation(), literalText));
+                        context.ReportDiagnostic(Diagnostic.Create(
+                            sourceNotStaticDescriptor,
+                            syntaxNode.GetLocation(),
+                            stringConstant));
+                    }
+
+                    switch (symbol)
+                    {
+                        case IPropertySymbol property:
+                            ReportIfSymbolNotIEnumerable(property.Type);
+                            ReportIfParametersSupplied(attributeInfo.NumberOfMethodParameters);
+                            break;
+                        case IFieldSymbol field:
+                            ReportIfSymbolNotIEnumerable(field.Type);
+                            ReportIfParametersSupplied(attributeInfo.NumberOfMethodParameters);
+                            break;
+                        case IMethodSymbol method:
+                            ReportIfSymbolNotIEnumerable(method.ReturnType);
+
+                            if (method.Parameters.Length != attributeInfo.NumberOfMethodParameters)
+                            {
+                                // TODO Report mismatch of parameters
+                            }
+
+                            break;
                     }
                 }
             }
         }
 
-        private static bool HasMember(SyntaxNodeAnalysisContext context, LiteralExpressionSyntax literal)
+        private static void ReportIfSymbolNotIEnumerable(ITypeSymbol typeSymbol)
         {
-            if (!SyntaxFacts.IsValidIdentifier(literal.Token.ValueText))
+            if (!typeSymbol.IsIEnumerable(out var _))
+            {
+                // TODO Report not IEnumerable
+            }
+        }
+
+        private static void ReportIfParametersSupplied(int? numberOfMethodParameters)
+        {
+            if (numberOfMethodParameters > 0)
+            {
+                // TODO Report parameters passed to field or property
+            }
+        }
+
+        private static SourceAttributeInformation ExtractInfoFromAttribute(
+            SyntaxNodeAnalysisContext context,
+            AttributeSyntax attributeSyntax)
+        {
+            var attributePositionalAndNamedArguments = attributeSyntax.GetArguments();
+            var positionalArguments = attributePositionalAndNamedArguments.Item1;

```

Comment:
Can we change this method to return ValueTuple?...

Reply:
We can. Actually, I tried to do it at some point, but I got some strange compile errors. I could blame my preview VS 😄 , but I think I just did something wrong. It is ValueTuple now (and I'll make another PR get rid of the remaining Item1, ...).